core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -847,7 +847,7 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
       }
       if (brokerIdGenerationEnable) {
         if (migrationEnabled) {
-          require(brokerId != -1, "broker id generation is incompatible with migration to ZK. Please disable it before enabling migration")
+          require(brokerId != -1, "broker.id generation is incompatible with ZooKeeper migration. Please stop using it before enabling migration (set broker.id to a value greater or equal to 0).")
```

Comment:
nit: I think we can set the condition explicitly to `brokerId >= 0` in this line. But of course, it's a pretty minor comment. It's fine we merge it as is for v3.9.0.

Reply:
I updated it to `>= 0`. It makes sense to use the same in all checks.